File: include/apl/errors/DomainError.java
package APL.errors;

import APL.types.*;

public class DomainError extends APLError {
  public DomainError(std::string s) {
    super(s);
  }
  public DomainError(std::string s, Tokenable fun) {
    super(s, fun);
  }
  public DomainError(std::string s, Callable fun, Tokenable cause) {
    super(s, fun, cause);
  }
}

File: APL/errors/DomainError.cpp
#include <APL/DomainError.h>

namespace APL::errors
{
    // using namespace APL::types;

    DomainError::DomainError(std::string const& s) : APLError(s)
    {
    }

    DomainError::DomainError(std::string const& s, std::shared_ptr<APL::types::Tokenable> fun) : APLError(s, fun)
    {
    }

    DomainError::DomainError(std::string const& s, std::shared_ptr<APL::types::Callable> fun, std::shared_ptr<APL::types::Tokenable> cause) : APLError(s, fun, cause)
    {
    }
}
